### ISSUE TITLE
Problems with GroovyTemplateEngineRenderingTest on JDK 11

### ISSUE THREAD

**Issue by ancho** [jbake-org/jbake#546]
While trying to fix an issue I noticed a problem executing `GroovyTemplateEngineRenderingTest` with JDK 11.

```
gradle -version
executing gradlew instead of gradle
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/home/frank/.gradle/wrapper/dists/gradle-4.10.2-bin/cghg6c4gf4vkiutgsab8yrnwv/gradle-4.10.2/lib/groovy-all-2.4.15.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

------------------------------------------------------------
Gradle 4.10.2
------------------------------------------------------------

Build time:   2018-09-19 18:10:15 UTC
Revision:     b4d8d5d170bb4ba516e88d7fe5647e2323d791dd

Kotlin DSL:   1.0-rc-6
Kotlin:       1.2.61
Groovy:       2.4.15
Ant:          Apache Ant(TM) version 1.9.11 compiled on March 23 2018
JVM:          11 (Oracle Corporation 11+28)
OS:           Linux 4.18.0-1-amd64 amd64
```

(You can ignore the warning. It's known, but not a serious Problem)

Anyways...

```
gradle clean test
executing gradlew instead of gradle
Starting a Gradle Daemon, 1 busy and 5 incompatible Daemons could not be reused, use --status for details

> Configure project :jbake-core
configured sonar url: https://sonarcloud.io
Skipping sonar analysis. Missing environment variables SONARLOGIN or SONARORG. See https://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner+for+Gradle

> Task :jbake-core:compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /home/frank/Projekte/j/jbake/git/jbake/jbake-core/src/main/java/org/jbake/parser/AsciidoctorEngine.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :jbake-core:compileTestJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /home/frank/Projekte/j/jbake/git/jbake/jbake-core/src/test/java/org/jbake/app/configuration/ConfigUtilTest.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :jbake-core:test
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.orientechnologies.common.directmemory.OByteBufferPool (file:/home/frank/.gradle/caches/modules-2/files-2.1/com.orientechnologies/orientdb-core/2.2.37/7a74182881316914fedd3d3afdd1c44cc5869b/orientdb-core-2.2.37.jar) to method sun.nio.ch.DirectBuffer.cleaner()
WARNING: Please consider reporting this to the maintainers of com.orientechnologies.common.directmemory.OByteBufferPool
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

org.jbake.app.template.GroovyTemplateEngineRenderingTest > renderIndex FAILED
    org.codehaus.groovy.GroovyBugError
        Caused by: java.lang.UnsupportedOperationException

org.jbake.app.template.GroovyTemplateEngineRenderingTest > renderArchive FAILED
    org.codehaus.groovy.GroovyBugError
        Caused by: java.lang.UnsupportedOperationException

org.jbake.app.template.GroovyTemplateEngineRenderingTest > checkDbTemplateModelIsPopulated FAILED
    org.codehaus.groovy.GroovyBugError
        Caused by: java.lang.UnsupportedOperationException

org.jbake.app.template.GroovyTemplateEngineRenderingTest > renderTagsIndex FAILED
    org.codehaus.groovy.GroovyBugError
        Caused by: java.lang.UnsupportedOperationException

org.jbake.app.template.GroovyTemplateEngineRenderingTest > renderFeed FAILED
    org.codehaus.groovy.GroovyBugError
        Caused by: java.lang.UnsupportedOperationException

org.jbake.app.template.GroovyTemplateEngineRenderingTest > renderPost FAILED
    org.codehaus.groovy.GroovyBugError
        Caused by: java.lang.UnsupportedOperationException

org.jbake.app.template.GroovyTemplateEngineRenderingTest > renderTags FAILED
    org.codehaus.groovy.GroovyBugError
        Caused by: java.lang.UnsupportedOperationException

214 tests completed, 7 failed

> Task :jbake-core:test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':jbake-core:test'.
> There were failing tests. See the report at: file:///home/frank/Projekte/j/jbake/git/jbake/jbake-core/build/reports/tests/test/index.html

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 50s
7 actionable tasks: 6 executed, 1 up-to-date
```

This is a known issue with groovy and JDK 11. 
See https://issues.apache.org/jira/browse/GROOVY-8727

It's fixed in 2.5.3. The [release vote](https://mail-archives.apache.org/mod_mbox/groovy-dev/201810.mbox/%3CCAMbkE7QEPXByttT2NtOUu-6k6OC6CpH2gqo6stmdna6Hd_VbOA%40mail.gmail.com%3E) has started.

**Comment by jonbullock** [comment]
Be good to get to get JBake working on JDK 11 👍

**Comment by jonbullock** [comment]
Fixed by #548
